src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.SslProvider.cs

```diff
@@ -180,8 +205,30 @@ private static int VerifyCertChain(IntPtr storeCtxPtr, IntPtr arg)
 
                         using (X509Certificate2 leafCert = new X509Certificate2(leafCertPtr))
                         {
-                            success = chain.Build(leafCert);
-                            AddChannelBindingToken(leafCert, arg);
+                            var serverCallback = easy._handler._serverCertificateValidationCallback;
+                            if (serverCallback == null)
+                            {
+                                SslPolicyErrors errors = CertificateValidation.BuildChainAndVerifyProperties(
+                                    chain, leafCert, checkCertName: false, hostName: null);
+                                success = errors == SslPolicyErrors.None;
```

Comment:
Having our X509CheckName used instead of theirs would allow for guaranteed consistency between no-callback and callback status codes.  So if we're going to do that we should just set CURLOPT_SSL_VERIFYHOST to 0 and always do it ourselves.

Or we could hope that they're the same logic, and only run our algorithm to build the SslPolicyErrors value.